api/src/pcapi/validation/routes/users_authentifications.py

```diff
@@ -93,15 +93,15 @@ def _get_current_api_key() -> ApiKey | None:
 
 
 def basic_authentication() -> User | None:
-    # `pcapi.utis.login_manager` cannot be imported at module-scope,
+    # `pcapi.utils.login_manager` cannot be imported at module-scope,
     # because the application context may not be available and that
     # module needs it.
     from pcapi.utils.login_manager import get_request_authorization
 
     auth = get_request_authorization()
     # According to the Werkzeug documentation auth.password is None
     # for any auth that is not basic auth.
-    if not auth or not auth.password:
+    if not auth or not auth.password or not auth.username:
```

Comment:
On va donc renvoyer None au lieu de lever l'exception `InvalidIdentifier` dans le cas où `auth.username` est nul. On est sûr que ça ne va rien casser (ça doit être marginal si c'est le cas mais bon) ?

Reply:
Hmm, bien vu. Je vais remplacer par `auth.username is None`. Si ça arrivait, ce cas lèverait une erreur dans `sanitize_email()`, appelé par `_find_user_by_email_query()`, appelé par `find_user_by_email()`, appelé par `get_user_with_credentials()` quelques lignes plus bas dans cette fonction. Donc je suppose que ça n'arrive jamais. J'ajoute cette condition pour aider mypy.